### PR TITLE
Integrate inline help toggle button to com_config component Options view (see #35610)

### PR DESCRIPTION
**Pull request by cyrez**
Pull Request to extend PR #35610.
cc/ @nikosdion 

### Summary of Changes
- Integrate inline help (toggle button) feature to com_config component Options view
- [com_content] : show inline help toggle button
- Other components will be updated if it's ok with this PR the way i did it. ;-)

### Testing Instructions
1. Apply Patch
2. Go to **Articles component**
3. Open Options
4. Inline help text hidden
5. Click on the toggle inline help button on top right to show / hide inline help.

### Actual result BEFORE applying this Pull Request
No inline help toggle button in com_content options.

### Expected result AFTER applying this Pull Request
Toggle button to show/hide inline help in com_content options.

### Documentation Changes Required
To integrate inline help toggle button in any component (core + third party), add the following line in your config.xml file for components. (after <help /> element)
`	<inlinehelp button="show" targetclass=""/>`

Params:
- button : 'show' to enable the toggle button for inline help text.
- targetclass : the class of the DIVs to toggle visibility on.

### CODE REVIEW

**Comment by prakhar3062** [comment]
branch is out of date  so not able to test it

**Comment by alikon** [comment]
you can use the  Prebuilt package

File: administrator/components/com_config/src/View/Component/HtmlView.php [hunk 1]
```diff
@@ -119,6 +119,14 @@ protected function addToolbar()
 		ToolbarHelper::cancel('component.cancel', 'JTOOLBAR_CLOSE');
 		ToolbarHelper::divider();
 
+		$inlinehelp  = (string) $this->form->getXml()->config->inlinehelp['button'] == 'show' ?: false;
```
Comment (Quy) [inline administrator/components/com_config/src/View/Component/HtmlView.php:122]:
```suggestion
		$inlinehelp  = (bool) $this->form->getXml()->config->inlinehelp['button'] == 'show' ?: false;
```

**Comment by nikosdion** [comment]
I have tested this item :white_check_mark: successfully on 80c2f8aa0a87923bcd8c0e92d01228af5a7b382a<br /><br />I am 100% onboard with this change. It will allow 3PDs to use this feature as well (which is something I am personally really looking forward to!).

As for com_content and other components, we will need to go through their Options pages with a fine comb to see which options need to be documented and how. But as you said that't the subject of another PR :)<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37158">issues.joomla.org/tracker/joomla-cms/37158</a>.</sub>

**Comment by cyrez** [comment]
> It will allow 3PDs to use this feature as well (which is something I am personally really looking forward to!).

Testing with my own extension, and working perfectly (even including your inlinehelp button in my edit view works well!)
I have maybe one suggestion: possibility to set by default visibility on show or hide (currently only hidden by default).
Storing in user session preference could be a bonus too.
This is a very useful feature, and allow better options documentation without throwing unnecessary text for advanced users.
Good work! 👍 

> As for com_content and other components, we will need to go through their Options pages with a fine comb to see which options need to be documented and how. But as you said that't the subject of another PR :)

So, should i enable it in all core components in this PR, or wait inlinehelp review for each component, and enable option when review is complete ?

**Comment by nikosdion** [comment]
@cyrezdev Regarding this:

> possibility to set by default visibility on show or hide (currently only hidden by default).
> Storing in user session preference could be a bonus too.

If you read the original PR (https://github.com/joomla/joomla-cms/pull/35610), namely the “Why make this change” you will see why this is not going to happen. You already figure it out yourself.

> This is a very useful feature, and allow better options documentation without throwing unnecessary text for advanced users.

Yup. That's why.

Ideally, we'd need real user testing to know if the assumption that beginners will click that button is true or if we need to revise. Fat chance of that happening.

The next best thing I could do is deploy support of Show Inline Help to my security software which has an infamously extensive configuration page. The past ~9 years it had tooltips. Now it has Show Inline Help. Within a few short months I will know if Show Inline Help works for inexperienced users. Until then, I am not fighting this battle no more! :D 

> So, should i enable it in all core components in this PR, or wait inlinehelp review for each component, and enable option when review is complete ?

Leave this PR as is. If it gets RTC and we get official word it's getting merged you can do a new PR for all other core components and _then_ we can then work on PRs for the help strings of each component (one per component) — the latter will need additional review from native English speakers to ensure clarity and conciseness.

**Comment by laoneo** [comment]
I have tested this item :white_check_mark: successfully on 80c2f8aa0a87923bcd8c0e92d01228af5a7b382a<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37158">issues.joomla.org/tracker/joomla-cms/37158</a>.</sub>

**Comment by alikon** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37158">issues.joomla.org/tracker/joomla-cms/37158</a>.</sub>

**Comment by cyrez** [comment]
> Ideally, we'd need real user testing to know if the assumption that beginners will click that button is true or if we need to revise. Fat chance of that happening.

For another PR: Maybe an alert information message on top of config to inform user of this new functionnality when available ?

Capture d’écran 2022-03-02 à 17 24 58

**Comment by bembelimen** [comment]
Thx

**Pull request merged**